webpack/assets/javascripts/react_app/components/factCharts/index.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Button, Modal } from 'react-bootstrap';
+import helpers from '../../common/helpers';
+import Chart from '../common/charts/Chart';
+import Loader from '../common/Loader';
+import MessageBox from '../common/MessageBox';
+import { getLargePieChartConfig, setTitle } from '../../../services/ChartService';
+import { STATUS } from '../../constants';
+import * as FactChartActions from '../../redux/actions/factCharts/';
+
+class FactChart extends React.Component {
+  constructor(props) {
+    super(props);
+    helpers.bindMethods(this, ['onClick', 'closeModal']);
+  }
+
+  onClick() {
+    const { showModal, getChartData, data: { id, path, title } } = this.props;
+
+    getChartData(path, id);
+    showModal(id, title);
+  }
+
+  closeModal() {
+    this.props.closeModal();
+  }
+
+  render() {
+    const { factChart, data: { id, title } } = this.props;
+
+    const showModal = factChart.displayModal[id];
+
+    const chartConfig = showModal ? getLargePieChartConfig({ data: factChart.chartData }) : false;
+
+    const chart = showModal ? (<Chart
+        className="c3-statistics-pie"
+        setTitle={setTitle}
+        config={chartConfig}
+        key={`chart-${id}`}
+      />) : false;
+
+    const requestErrorMsg = factChart.loaderStatus === STATUS.ERROR ? __('Request Failed') : undefined;
+
+    const error = showModal ? (
+        <MessageBox
+          msg={ requestErrorMsg || __('No data available') }
+          icontype="error-circle-o"
+          key={`message-${id}`}
+        />) : false;
+
+    return (<div>
+        <Button onClick={this.onClick}>{__('View Chart')}</Button>
+        {showModal ?
+          <Modal show={true} onHide={this.closeModal}>
+            <Modal.Header>
+              <Modal.Title>
+                <b>
+                  {
+                    // eslint-disable-next-line no-undef
+                    Jed.sprintf(__('Fact distribution chart - %s'), title)
+                  }
+                </b>
+              </Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+              <Loader status={factChart.loaderStatus}>{[chart, error]}</Loader>
+            </Modal.Body>
+            <Modal.Footer>
+              <Button onClick={this.closeModal}>{__('Close')}</Button>
+            </Modal.Footer>
+          </Modal>
+          :
+          false
+        }
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state, ownProps) => ({
+  factChart: ownProps.factChart || state.factChart,
```

Comment:
You can map the specific dispalyModal state to props, something like this:
```javascript
displayModal: state.factChart.displayModal[ownProps.id] || false,
```
after this change,  line 32 becomes redundant